Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocLargeEventCollectionViewCell.swift

```diff
@@ -228,7 +232,22 @@ class ArticleAsLivingDocLargeEventCollectionViewCell: CollectionViewCell {
         viewChangesButton.backgroundColor = theme.colors.midBackground
         viewDiscussionButton.setTitleColor(theme.colors.link, for: .normal)
         viewDiscussionButton.backgroundColor = theme.colors.midBackground
-        
+
+        if let largeEvent = largeEvent {
+            switch largeEvent.buttonsToDisplay {
+            case .thankAndViewChanges:
+                thankButton.backgroundColor = largeEvent.wereThanksSent ? theme.colors.hintBackground : theme.colors.midBackground
```

Comment:
I believe I fixed the issue by removing [these lines](https://github.com/wikimedia/wikipedia-ios/pull/3736/commits/b7c3501771057b14c75a4aed1afdaa63add79ca2#diff-b2d1ab9c419802357bea7f6596cb243809286c78fff206bfeba00f6cc0628540L213-L218), which were blocking a theme refresh from happening. (We use the theme refresh to handle updating the buttons appropriately.)

@staykids - can you replicate this? I can't replicate at this point, but I wasn't able to get all of the misfires you reported above.

@tonisevener - I believe you wrote the lines I removed. Was it to help with performance? Or did I cause some regression by removing those, and need to solve this another way?

Reply:
@mcleinman ah good catch. Yeah I originally did that to help with performance so that attributed strings aren't reset and rebuilt unnecessarily, but after digging further it looks like I was already doing this check elsewhere within the reset/rebuild methods themselves. It should be fine taking this out.

Reply:
I'll keep poking but so far looks like that resolved it!